tests/gatewayapitests/graphlayer/httproute_test.go

```diff
@@ -1398,7 +1397,155 @@ func TestHTTPRouteWithMultipleGateways(t *testing.T) {
+	akogatewayapitests.UpdateHTTPRoute(t, httpRouteName1, DEFAULT_NAMESPACE, []gatewayv1.ParentReference{parentRefs[0]}, hostnames, rules)
+
+	g.Eventually(func() int {
+		_, aviModel1 = objects.SharedAviGraphLister().Get(modelName)
+		nodes = aviModel1.(*avinodes.AviObjectGraph).GetAviEvhVS()
+		return len(nodes[0].VSVIPRefs[0].FQDNs)
+	}, 25*time.Second, 1*time.Second).Should(gomega.Equal(2))
+
+	// validate parent
+	_, aviModel1 = objects.SharedAviGraphLister().Get(modelName)
+	nodes = aviModel1.(*avinodes.AviObjectGraph).GetAviEvhVS()
+	g.Expect(len(nodes[0].VSVIPRefs[0].FQDNs)).To(gomega.Equal(2))
+	g.Expect(nodes[0].VSVIPRefs[0].FQDNs[0]).To(gomega.Equal("abc.avi.internal"))
+	g.Expect(nodes[0].VSVIPRefs[0].FQDNs[1]).To(gomega.Equal("efg.avi.internal"))
+
+	// update httproute with new hostname
+	hostnames = []gatewayv1.Hostname{"abcdef.avi.internal", "efg.avi.internal"}
+	akogatewayapitests.UpdateHTTPRoute(t, httpRouteName1, DEFAULT_NAMESPACE, []gatewayv1.ParentReference{parentRefs[0]}, hostnames, rules)
+
+	g.Eventually(func() string {
+		found, aviModel := objects.SharedAviGraphLister().Get(modelName)
+		if !found {
+			return ""
+		}
+		nodes := aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+		return nodes[0].VSVIPRefs[0].FQDNs[0]
+	}, 25*time.Second).Should(gomega.Equal("abcdef.avi.internal"))
+
+	// check other parent fqdns
+	_, aviModel1 = objects.SharedAviGraphLister().Get(modelName)
+	nodes = aviModel1.(*avinodes.AviObjectGraph).GetAviEvhVS()
+	g.Expect(len(nodes[0].VSVIPRefs[0].FQDNs)).To(gomega.Equal(2))
+	g.Expect(nodes[0].VSVIPRefs[0].FQDNs[0]).To(gomega.Equal("abcdef.avi.internal"))
+	g.Expect(nodes[0].VSVIPRefs[0].FQDNs[1]).To(gomega.Equal("efg.avi.internal"))
+
+	// create httproute 2 and attach it to listener 2
+	rule1 = akogatewayapitests.GetHTTPRouteRuleV1([]string{"/foo"}, []string{}, nil,
+		[][]string{{svcName, DEFAULT_NAMESPACE, "8080", "1"}}, nil)
+
+	rules = []gatewayv1.HTTPRouteRule{rule1}
+	hostnames = []gatewayv1.Hostname{"hello.avi.internal"}
+
+	akogatewayapitests.SetupHTTPRoute(t, httpRouteName2, DEFAULT_NAMESPACE, []gatewayv1.ParentReference{parentRefs[1]}, hostnames, rules)
+	g.Eventually(func() int {
+		found, aviModel := objects.SharedAviGraphLister().Get(modelName)
+		if !found {
+			return -1
+		}
+		nodes := aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+		return len(nodes[0].EvhNodes)
+	}, 25*time.Second).Should(gomega.Equal(2))
+
+	// validate parent
+	_, aviModel1 = objects.SharedAviGraphLister().Get(modelName)
+	nodes = aviModel1.(*avinodes.AviObjectGraph).GetAviEvhVS()
+	g.Expect(len(nodes[0].PoolGroupRefs)).To(gomega.Equal(0))
+	g.Expect(len(nodes[0].VSVIPRefs[0].FQDNs)).To(gomega.Equal(3))
+	g.Expect(nodes[0].VSVIPRefs[0].FQDNs[0]).To(gomega.Equal("abcdef.avi.internal"))
+	g.Expect(nodes[0].VSVIPRefs[0].FQDNs[1]).To(gomega.Equal("efg.avi.internal"))
+	g.Expect(nodes[0].VSVIPRefs[0].FQDNs[2]).To(gomega.Equal("hello.avi.internal"))
+
```

Comment:
Please add one more update here:
Try deleting one of the hostname from route 1 and then see if there are resulting 2 hostnames on parentvs

Reply:
done